From 0833601fa95295d21256d0c330fb0b43adf755b5 Mon Sep 17 00:00:00 2001
From: Xie XiuQi <xiexiuqi@huawei.com>
Date: Fri, 26 Feb 2021 20:21:30 +0800
Subject: [PATCH] arm64/mpam: remove unsupported resource
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: 0833601fa95295d21256d0c330fb0b43adf755b5


hulk inclusion
category: bugfix
bugzilla: 48265
CVE: NA

--------------------------------

We just support L3 & Memory bandwidth on this version,
so remove unsported resource type.

Signed-off-by: Xie XiuQi <xiexiuqi@huawei.com>
Reviewed-by: Hanjun Guo <guohanjun@huawei.com>
Signed-off-by: Yang Yingliang <yangyingliang@huawei.com>
Signed-off-by: Wang ShaoBo <bobo.shaobowang@huawei.com>
Reviewed-by: Cheng Jian <cj.chengjian@huawei.com>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 arch/arm64/kernel/mpam.c | 16 ----------------
 1 file changed, 16 deletions(-)

diff --git a/arch/arm64/kernel/mpam.c b/arch/arm64/kernel/mpam.c
index c38c13bde995..a9dab7eb7b77 100644
--- a/arch/arm64/kernel/mpam.c
+++ b/arch/arm64/kernel/mpam.c
@@ -172,14 +172,6 @@ static int csu_write(struct rdt_domain *d, struct rdtgroup *g, bool enable);
 #define domain_init(id) LIST_HEAD_INIT(resctrl_resources_all[id].domains)
 
 struct raw_resctrl_resource raw_resctrl_resources_all[] = {
-	[MPAM_RESOURCE_SMMU] = {
-		.msr_update		= cat_wrmsr,
-		.msr_read		= cat_rdmsr,
-		.parse_ctrlval		= parse_cbm,
-		.format_str		= "%d=%0*x",
-		.mon_read		= csu_read,
-		.mon_write		= csu_write,
-	},
 	[MPAM_RESOURCE_CACHE] = {
 		.msr_update		= cat_wrmsr,
 		.msr_read		= cat_rdmsr,
@@ -199,14 +191,6 @@ struct raw_resctrl_resource raw_resctrl_resources_all[] = {
 };
 
 struct resctrl_resource resctrl_resources_all[] = {
-	[MPAM_RESOURCE_SMMU] = {
-		.rid			= MPAM_RESOURCE_SMMU,
-		.name			= "SMMU",
-		.domains		= domain_init(MPAM_RESOURCE_SMMU),
-		.res			= &raw_resctrl_resources_all[MPAM_RESOURCE_SMMU],
-		.fflags			= RFTYPE_RES_SMMU,
-		.alloc_enabled		= 1,
-	},
 	[MPAM_RESOURCE_CACHE] = {
 		.rid			= MPAM_RESOURCE_CACHE,
 		.name			= "L3",
-- 
2.26.2

